### ISSUE TITLE
Accessing string with percentage symbol strips it

### ISSUE THREAD

**Issue by igorkulman** [SwiftGen/SwiftGen#859]
I am using SwiftGen to generate strings with the structured swift5 template

```yaml
strings:
  inputs:
    - Source/Localization/en.lproj/Localizable.strings
    - Source/Localization/en.lproj/Localizable.stringsdict
  outputs:
    params:
      publicAccess: true
      enumName: LocalizedString
    templateName: structured-swift5
    output: Source/LocalizedString.swift
```

If I have a string that contains the percentage symbol, like

```
"percentage_symbol_test" = "Result is 100%";
```

accessing `LocalizedString.percentageSymbolTest` gives me `Result is 100` with the percentage being stripped.

**Comment by harryblam** [comment]
I'm seeing this issue as well. 

I've found a workaround however, which i'm using whenever I need to use the % symbol in my strings. I simple substitute the symbol with a `%@` placeholder, which allows me to input the % symbol as a string in code. 
I will give an example using the above code:
```"percentage_symbol_workaround" = "Result is %@";```

and the usage in Swift code:
```L10n.percentageSymbolWorkaround("100%");```

**Comment by igorkulman** [comment]
My workaround us doubling the percentage symbol, so `"percentage_symbol_test" = "Result is 100%%";
`

**Comment by djbe** [comment]
Duplicate of quite a few issues:
- String with % (100% for example) #831
- Strings: % formatting #577
- Problems with '%' symbol #460

This is an effect of **all** translations being run through `String(format:)`. At some point we tried being "smarter" about this, i.e. detect if there were any parameters in the translation, and if not, to not run it through `String(format:)`, but that led to a whole bunch of bugs for other users, so we quickly had to revert it.

**TL;DR**: you have to escape the `%`, so your string becomes: `Result is 100%%`.

**Comment by harryblam** [comment]
> Duplicate of quite a few issues:
> 
> * String with % (100% for example) [String with % (100% for example) #831](https://github.com/SwiftGen/SwiftGen/issues/831)
> * Strings: % formatting [Strings: % formatting #577](https://github.com/SwiftGen/SwiftGen/issues/577)
> * Problems with '%' symbol [Problems with '%' symbol #460](https://github.com/SwiftGen/SwiftGen/issues/460)
> 
> This is an effect of **all** translations being run through `String(format:)`. At some point we tried being "smarter" about this, i.e. detect if there were any parameters in the translation, and if not, to not run it through `String(format:)`, but that led to a whole bunch of bugs for other users, so we quickly had to revert it.
> 
> **TL;DR**: you have to escape the `%`, so your string becomes: `Result is 100%%`.

Thanks @djbe for the info! It makes sense now that you've explained the `String(format:)` side effects. Perhaps it would be worth updating the README with this info as "gotcha" or something? 

I guess I could have found this out by searching the closed issues but I don't normally go to that much effort before raising an issue tbh 😅 .
